src/main/java/cz/datadriven/utils/config/view/ConfigViewProxy.java

```diff
@@ -46,8 +50,14 @@
           ConfigView.Bytes.class,
           ConfigView.Map.class);
 
-  static class Factory {
+  static class Factory implements Serializable {
+
+    private static final long serialVersionUID = 62698747501317112L;
 
+    @SuppressFBWarnings(
+        value = "SE_BAD_FIELD",
+        justification =
+            "Config interface doesn't extend Serializable, but the non-public implementation SimpleConfig does")
     private final Config config;
```

Comment:
can we make the config transient and serialize string representation instead?

```java
final String str = config.root().render(ConfigRenderOptions.concise());
ConfigFactory.parseString(str);
```

Reply:
or maybe better, we can do something like this (untested):

```
import com.typesafe.config.Config;
import com.typesafe.config.ConfigFactory;
import com.typesafe.config.ConfigRenderOptions;
import java.io.Serializable;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;

public class SerializableConfig implements Serializable {

  private static final long serialVersionUID = 1L;

  private static class SerializedConfig implements Serializable {

    private static final long serialVersionUID = 1L;
    private static final ConcurrentMap<String, SerializableConfig> cache =
        new ConcurrentHashMap<>();

    private final String rawConfig;

    SerializedConfig(String rawConfig) {
      this.rawConfig = rawConfig;
    }

    protected Object readResolve() {
      return cache.computeIfAbsent(
          rawConfig, key -> new SerializableConfig(ConfigFactory.parseString(key)));
    }
  }

  private final Config config;

  public SerializableConfig(Config config) {
    this.config = config;
  }

  public Config get() {
    return config;
  }

  protected Object writeReplace() {
    return new SerializedConfig(config.root().render(ConfigRenderOptions.concise()));
  }
}
```

Reply:
ok, I implemented the second proposal

Reply:
could we make it explicitly serializable now?

Reply:
I really think that user should always define wheter it's serializable or not, so I'd rather avoid it.